samples/sample_vpp/src/sample_vpp_config.cpp

```diff
@@ -131,6 +131,13 @@ mfxStatus ConfigVideoEnhancementFilters( sInputParams* pParams, sAppResources* p
         pVppParam->ExtParam[pVppParam->NumExtParam++] = (mfxExtBuffer*)&(pResources->mirroringConfig);
     }
 
+    if( VPP_FILTER_ENABLED_CONFIGURED == pParams->colorfillParam[paramID].mode )
+    {
+        pResources->colorfillConfig = pParams->colorfillParam[paramID];
+        pVppParam->ExtParam[pVppParam->NumExtParam++] = (mfxExtBuffer*)&(pResources->colorfillConfig);
```

Comment:
Instead of (mfxExtBuffer*)&(pResources->colorfillConfig) use &(pResources->colorfillConfig.Header)

Reply:
same

Reply:
AFAIR, ExtParam should keep pointer to the external buffer structure, so it's more correct to use (mfxExtBuffer*)&(pResources->colorfillConfig) However, there's no difference as long as Header is the first element in buffer (and in memory)

Reply:
Header type is exactly mfxExtBuffer, it is similar to base class. It was designed to be used this way. Casting can hide possible errors.
Again, fix it only in modified code.

Reply:
agree with @lakulako. Try to avoid excessive casts

Reply:
done